core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -102,6 +106,26 @@ case class Uri(
   override protected def self: Self = this
 
   override protected def replaceQuery(query: Query): Self = copy(query = query)
+
+  /**
+   * Converts a [[Uri]] into an [[InetSocketAddress]] as long as this [[Uri]] has an authority.
+   * This conversion is a pretty straightforward translation, most of the work is inserting default port values
+   * if the [[Uri]] does not specify explicit ports.
+   * This method only has knowledge of the Http protocol (including websockets), it does not know how to guess ports
+   * for schemes it has no knowledge of and will return an [[IOException]] if it encounters an unknown scheme.
+   * If the uri does not explicit a scheme, it will assume the [[Http]] scheme.
+   */
+  def asAddress: Throwable \/ InetSocketAddress = {
```

Comment:
This function is broken: it fails to use a port that is explicitly specified. The following works for me:
```scala
def asAddress: Throwable \/ InetSocketAddress = {
    authority.map { auth =>
      val host = auth.host.value
      auth.port.map(port => \/-(new InetSocketAddress(host, port))).getOrElse {
        val assumedScheme = scheme.getOrElse(Http)
        assumedScheme match {
          case Http | Ws                => \/-(new InetSocketAddress(host, 80))
          case Https | Wss              => \/-(new InetSocketAddress(host, 443))
          case _                        => -\/(new IOException("Unknown scheme"))
        }
      }
    }.getOrElse(-\/(new IOException("Request must have an authority")))
  }
```